### PR TITLE
Make QueueIterator::isValid() const and fix typo

### PR DESCRIPTION
**Pull request by avinabadasgupta**

### CODE REVIEW

File: sparta/sparta/resources/Queue.hpp [hunk 1]
```diff
@@ -329,28 +329,28 @@ namespace sparta
...
-                sparta_assert(physical_idx_ < attatched_queue_->vector_size_, "Not a valid Queue Iterator" );
-                return attatched_queue_->convertPhysicalIndex_(physical_idx_);
+                sparta_assert(physical_idx_ < attached_queue->vector_size_, "Not a valid Queue Iterator" );
+                return attached_queue->convertPhysicalIndex_(physical_idx_);
             }
 
         private:
 
-            QueuePointerType attatched_queue_;
+            QueuePointerType attached_queue;
```
Comment (knute-sifive) [inline sparta/sparta/resources/Queue.hpp]: Why did you drop the trailing `_`?

Comment (avinabadasgupta) [reply sparta/sparta/resources/Queue.hpp]: Sorry, find and replace error. Adding it back.

**Comment by knute-sifive** [comment]
Thanks, Avi for getting on this one!

**Comment by avinabadasgupta** [comment]
> Thanks, Avi for getting on this one!

no problem, merging it in.

**Pull request merged**